ddl/ddl_api.go

```diff
@@ -1035,6 +1035,17 @@ func (d *ddl) handleAutoIncID(tbInfo *model.TableInfo, schemaID int64) error {
 	return nil
 }
 
+func setDefaultTableCharsetAndCollation(tbInfo *model.TableInfo) (err error) {
+	if len(tbInfo.Charset) == 0 {
```

Comment:
1. Why check `tbInfo.Charset` and `tbInfo.Collate` in this function?
2. It never return error? then the function signature can be changed.

Reply:
2. it return error in line 1044

Reply:
Can not understand your first question.

Reply:
From the function name `setDefaultTableCharsetAndCollation` I guess the `tbInfo.Charset` and `tbInfo.Collate` would be set unconditionally. so `if len(tbInfo.Charset)` confused me.
The real thing this function do is "setDefaultTableCharsetAndCollation" **if it's not done yet** .
That ok.